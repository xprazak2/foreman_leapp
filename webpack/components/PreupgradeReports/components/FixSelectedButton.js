import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';

const FixSelectedButton = ({ ids, postUrl, disabled, csrfToken }) => {
  const { hostIds, entryIds } = ids;

  return (
    <form action={postUrl} method="post">
      <Button type="submit" disabled={disabled}>
        {__('Fix Selected')}
      </Button>
      <input type="hidden" name="authenticity_token" value={csrfToken} />
      <input type="hidden" name="feature" value="leapp_remediation_plan" />
      {hostIds.map(hostId => (
        <input type="hidden" name="host_ids[]" key={hostId} value={hostId} />
      ))}
      <input
        type="hidden"
        name="inputs[remediation_ids]"
        value={entryIds.join(',')}
      />
    </form>
  );
};

FixSelectedButton.propTypes = {
  ids: PropTypes.object.isRequired,
  postUrl: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  csrfToken: PropTypes.string.isRequired,
};

export default FixSelectedButton;
