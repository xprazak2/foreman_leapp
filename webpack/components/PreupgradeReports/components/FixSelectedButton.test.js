import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import FixSelectedButton from './FixSelectedButton';

const fixtures = {
  'should render': {
    ids: { hostIds: [5], entryIds: [115] },
    postUrl: '/job_invocations/new',
    disabled: false,
    csrfToken: 'abcd',
  },
};

describe('FixSelectedButton', () =>
  testComponentSnapshotsWithFixtures(FixSelectedButton, fixtures));
