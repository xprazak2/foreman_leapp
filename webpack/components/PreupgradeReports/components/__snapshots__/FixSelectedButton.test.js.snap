// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixSelectedButton should render 1`] = `
<form
  action="/job_invocations/new"
  method="post"
>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    type="submit"
  >
    Fix All
  </Button>
  <input
    name="authenticity_token"
    type="hidden"
    value="abcd"
  />
  <input
    name="feature"
    type="hidden"
    value="leapp_remediation_plan"
  />
  <input
    key="5"
    name="host_ids[]"
    type="hidden"
    value={5}
  />
  <input
    name="inputs[remediation_ids]"
    type="hidden"
    value="115"
  />
</form>
`;
